# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsIFrameInlines.h
# Commit: e90eb44c2da3
# Full Hash: e90eb44c2da3303c038c67b7ad28b1498cd5a98c
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2024-07-19 16:21:39
# Description:
#   Bug 1897255 - Make `nsIFrame::PeekOffsetForLine` won't cross editing host boundary unless the callers allows that r=emilio
#   
#   The problems in the testcase are, `nsIFrame::PeekOffsetForLine` returns a frame
#   for any content node outside the editing host and `nsIFrame::GetLastLeaf`
#   returns a native anonymous subtree node if the frame is for the native
# ==============================================================================

diff -r bd02e7030721 -r e90eb44c2da3 layout/generic/nsIFrameInlines.h
--- a/layout/generic/nsIFrameInlines.h	Fri Jul 19 07:46:19 2024 +0000
+++ b/layout/generic/nsIFrameInlines.h	Fri Jul 19 08:13:10 2024 +0000
@@ -10,6 +10,7 @@
 #include "mozilla/dom/ElementInlines.h"
 #include "mozilla/ComputedStyleInlines.h"
 #include "nsContainerFrame.h"
+#include "nsIContentInlines.h"
 #include "nsLayoutUtils.h"
 #include "nsPlaceholderFrame.h"
 #include "nsCSSAnonBoxes.h"
@@ -186,4 +187,8 @@
                                aContainerSize - mRect.Size());
 }
 
+bool nsIFrame::ContentIsEditable() const {
+  return mContent && mContent->IsEditable();
+}
+
 #endif